packages/wrangler/src/type-generation.ts

```diff
@@ -0,0 +1,171 @@
+	rules: Config["rules"];
+};
+export async function generateTypes(
+	configToDTS: PartialConfigToDTS,
+	config: Config
+) {
+	const entry = await getEntry({}, config, "types");
+	const envTypeStructure: string[] = [];
+
+	if (configToDTS.kv_namespaces) {
+		for (const kvNamespace of configToDTS.kv_namespaces) {
+			envTypeStructure.push(`	${kvNamespace.binding}: KVNamespace;`);
+		}
+	}
+
+	if (configToDTS.vars) {
+		for (const varName in configToDTS.vars) {
+			const varValue = configToDTS.vars[varName];
+			if (
+				typeof varValue === "string" ||
+				typeof varValue === "number" ||
+				typeof varValue === "boolean"
+			) {
+				envTypeStructure.push(`	${varName}: ${varValue};`);
+			}
+			if (typeof varValue === "object" && varValue !== null) {
+				envTypeStructure.push(`	${varName}: ${JSON.stringify(varValue)};`);
+			}
+		}
+	}
+
+	if (configToDTS.durable_objects?.bindings) {
+		for (const durableObject of configToDTS.durable_objects.bindings) {
+			envTypeStructure.push(`	${durableObject.name}: DurableObjectNamespace;`);
+		}
+	}
+
+	if (configToDTS.r2_buckets) {
+		for (const R2Bucket of configToDTS.r2_buckets) {
+			envTypeStructure.push(`	${R2Bucket.binding}: R2Bucket;`);
+		}
+	}
+
+	if (configToDTS.d1_databases) {
+		for (const d1 of configToDTS.d1_databases) {
+			envTypeStructure.push(`	${d1.binding}: D1Database;`);
+		}
+	}
+
+	if (configToDTS.services) {
+		for (const service of configToDTS.services) {
+			envTypeStructure.push(`	${service.binding}: Fetcher;`);
+		}
+	}
+
+	if (configToDTS.dispatch_namespaces) {
+		for (const namespace of configToDTS.dispatch_namespaces) {
+			envTypeStructure.push(`	${namespace.binding}: any;`);
+		}
+	}
+
+	if (configToDTS.logfwdr?.schema) {
+		envTypeStructure.push(`	LOGFWDR_SCHEMA: any;`);
+	}
+
+	if (configToDTS.data_blobs) {
+		for (const dataBlobs in configToDTS.data_blobs) {
+			envTypeStructure.push(`	${dataBlobs}: ArrayBuffer;`);
+		}
+	}
+
+	if (configToDTS.text_blobs) {
+		for (const textBlobs in configToDTS.text_blobs) {
+			envTypeStructure.push(`	${textBlobs}: string;`);
+		}
+	}
+
+	if (configToDTS.unsafe) {
+		for (const unsafe of configToDTS.unsafe) {
+			envTypeStructure.push(`	${unsafe.name}: any;`);
+		}
+	}
+
+	const modulesTypeStructure: string[] = [];
+	if (configToDTS.rules) {
+		const typeToTypeScriptType = {
+			Text: "string",
+			Data: "ArrayBuffer",
+			CompiledWasm: "WebAssembly.Module",
+		};
+		for (const ruleObject of configToDTS.rules) {
+			const typeScriptType =
+				typeToTypeScriptType[
+					ruleObject.type as keyof typeof typeToTypeScriptType
+				];
+			if (typeScriptType !== undefined) {
+				ruleObject.globs.forEach((glob) => {
+					modulesTypeStructure.push(`declare module "${glob}" {
+	const value: ${typeScriptType};
+	export default value;
+}`);
+				});
+			}
+		}
+	}
+
+	function writeDTSFile(
+		typesString: string[],
+		formatType: "modules" | "service-worker"
+	) {
+		const wranglerOverrideDTSPath = findUpSync("wrangler-overrides.d.ts");
```

Comment:
Bikeshedding, but can we call this something different? Maybe `configuration.d.ts`? Or `cloudflare.d.ts`?

Reply:
Definitely not `configuration.d.ts` that is far too generic. Why `cloudflare.d.ts` over `wrangler-overrides.d.ts`?

Reply:
Because they aren't overrides (what would they be overriding?). Maybe `worker-configuration.d.ts`?

Reply:
Hmmm, yeah it's not really an override. I like this suggestion `worker-configuration.d.ts`